src/Mod/PartDesign/App/FeatureSketchBased.cpp

```diff
@@ -628,12 +636,56 @@ void ProfileBased::generatePrism(TopoDS_Shape& prism,
                 throw Base::ValueError("Cannot create a pocket with a depth of zero.");
         }
 
-        // Its better not to use BRepFeat_MakePrism here even if we have a support because the
-        // resulting shape creates problems with Pocket
-        BRepPrimAPI_MakePrism PrismMaker(from, Ltotal * gp_Vec(dir), 0, 1); // finite prism
-        if (!PrismMaker.IsDone())
-            throw Base::RuntimeError("ProfileBased: Length: Could not extrude the sketch!");
-        prism = PrismMaker.Shape();
+        if (fabs(angle) > Precision::Angular() || fabs(angle2) > Precision::Angular()) {

```

Comment:
There is still no point in having all this inside generatePrism(). Revert the changes and add two additional functions:
```
bool hasTaperedAngle(double angle1, double angle2) const {
   return fabs(angle) > Precision::Angular() || fabs(angle2) > Precision::Angular();
}
```
```
void ProfileBased::generateTaperedPrism(...)
{
   // move all the code from inside the if-block
}
```

The code in Extrusion::execute() then will become:
```
if (hasTaperedAngle(TaperAngle.getValue(), TaperAngle2.getValue())) {
    generateTaperedPrism(...);
}
else {
    generatePrism(...);
}
```

Reply:
> There is still no point in having all this inside generatePrism(). Revert the changes and add two additional functions:

Thanks, I will do so.

Reply:
Please note that `Precision::Angular()` returns a limit in radians, you seem to compare it with degrees.

Reply:
> Please note that Precision::Angular() returns a limit in radians, you seem to compare it with degrees.

Good catch! Fixed now.